server/src/main/scala/controllers/application/exports/CsvFileStreaming.scala

```diff
@@ -3,42 +3,35 @@ package controllers.application.exports
 import akka.NotUsed
 import akka.stream.scaladsl.Source
 import akka.util.ByteString
-import play.api.http.{HttpChunk, HttpEntity, Writeable}
+import play.api.http.{HttpEntity, Writeable}
 import play.api.mvc.{ResponseHeader, Result}
 import uk.gov.homeoffice.drt.ports.PortCode
 import uk.gov.homeoffice.drt.ports.Terminals.Terminal
-import uk.gov.homeoffice.drt.time.{LocalDate, SDate}
 import uk.gov.homeoffice.drt.time.TimeZoneHelper.europeLondonTimeZone
+import uk.gov.homeoffice.drt.time.{LocalDate, SDate}
 
 object CsvFileStreaming {
 
   def csvFileResult(fileName: String, data: String): Result = Result(
     ResponseHeader(200, Map("Content-Disposition" -> s"attachment; filename=$fileName.csv")),
     HttpEntity.Strict(ByteString(data), Option("application/csv")))
 
-  def sourceToCsvResponse(exportSource: Source[String, NotUsed], fileName: String): Result = {
-    val writeable: Writeable[String] = Writeable((str: String) => ByteString.fromString(str), Option("text/csv"))
+  def sourceToCsvResponse(exportSource: Source[String, NotUsed], fileName: String): Result =
+    streamingResponse(exportSource, "text/csv", Option(fileName))
 
-    Result(
-      header = ResponseHeader(200, Map("Content-Type" -> "text/csv", "Content-Disposition" -> s"attachment; filename=$fileName.csv")),
-      body = HttpEntity.Chunked(exportSource.collect {
-        case s if s.nonEmpty => s
-      }.map(c => {
-        HttpChunk.Chunk(writeable.transform(c))
-      }), writeable.contentType))
-  }
+  def sourceToJsonResponse(exportSource: Source[String, NotUsed]): Result =
+    streamingResponse(exportSource, "application/json", None)
+
+  private def streamingResponse(exportSource: Source[String, NotUsed], mimeType: String, maybeFileName: Option[String]): Result = {
+    val writeable: Writeable[String] = Writeable((str: String) => ByteString.fromString(str), Option(mimeType))
 
-  def sourceToJsonResponse(exportSource: Source[String, NotUsed]): Result = {
-    val writeable: Writeable[String] = Writeable((str: String) => ByteString.fromString(str), Option("application/json"))
+    val byteStringStream = exportSource.collect {
+      case s if s.nonEmpty => writeable.transform(s)
+    }
 
     Result(
-      header = ResponseHeader(200, Map("Content-Type" -> "application/json")),
-      body = HttpEntity.Chunked(
-        exportSource.collect {
-          case s if s.nonEmpty => HttpChunk.Chunk(writeable.transform(s))
-        },
-        writeable.contentType
-      )
+      header = ResponseHeader(200, Map("Content-Type" -> "application/json") ++ maybeFileName.map(fn => "Content-Disposition" -> s"attachment; filename=$fn.csv")),
```

Comment:
file name is csv and contenttype is json ?

Reply:
Good point! I didn't mean to leave the content type hard coded. Fixed now, and the file name doesn't have csv hard coded now either. It has to be supplied with the extension